Add setup.py to the Cache Keys for the GitHub Actions Workflow

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This adds `setup.py` to the files hashed to create the cache keys used in our GitHub Actions workflow. I am intentionally _not_ using `**/setup.py` so only the `setup.py` in the root directory of the repository (which should be the base path for `GITHUB_WORKSPACE` when running an Action) is used. This prevents unexpected behavior if a package contains a `setup.py`.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Since we direct requirements for our Python projects to be stored in `setup.py` we should invalidate a cache based on changes to that file. The file generally will not change much outside of requirements changes once a project is past it's initial implementation.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
I confirmed that in the GHA run the cache key is updated to include a hash for `setup.py`.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## 📷 Screenshots (if appropriate) ##

<!-- Remove this section and header if not needed -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

mcdonnnj committed:
Add to the cache keys for the GitHub Actions workflow

This adds a hash of `setup.py` to the cache keys used in the GitHub Actions
workflow for Python projects.

dav3r reviewed (approved):
Good improvement, though I'd recommend adding a comment explaining why `setup.py` is not `**/setup.py`.

mcdonnnj committed:
Add comments explaining different hashFiles() argument

The other cache keys for our GHA jobs are in the format '**/<filename>' so that
any file with that name is used in the repository. However, for Python packages
they may have a 'setup.py' as part of their internal codebase that does not
impact environment requirements. As a result we only want to use the 'setup.py'
that is in the root of the repository and is used to install the package.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj commented:
> Good improvement, though I'd recommend adding a comment explaining why `setup.py` is not `**/setup.py`.

Excellent suggestion. I've added a comment to every cache key declaration in 449eef4. Please let me know if you have any feedback about the comment.

dav3r commented:
> > Good improvement, though I'd recommend adding a comment explaining why `setup.py` is not `**/setup.py`.
> 
> Excellent suggestion. I've added a comment to every cache key declaration in [449eef4](https://github.com/cisagov/skeleton-python-library/commit/449eef422bf7cb8469a25996126fac26ae899fb4). Please let me know if you have any feedback about the comment.

Looks perfect to me - thanks for adding that.

mcdonnnj committed:
Merge branch 'develop' into improvement/add_setup_to_gha_cache_key
